src/cwe_checker_lib/src/intermediate_representation/expression/tests.rs

```diff
@@ -539,3 +539,22 @@ fn processing_sub_registers() {
     expr.cast_sub_registers_to_base_register_subpieces(output, &register_map, peeked);
     assert_eq!(expr, setup.int_sub_subpiece_expr);
 }
+
+#[test]
+fn display() {
+    let expr = Expression::const_from_i32(2);
+    let expr = expr.plus(Setup::new().int_sub_expr);
+    let expr = Expression::UnOp {
+        op: UnOpType::IntNegate,
+        arg: Box::new(expr),
+    };
+    let expr = expr
+        .cast(CastOpType::IntSExt)
+        .un_op(UnOpType::FloatCeil)
+        .subpiece(ByteSize(0), ByteSize(20));
+
+    assert_eq!(
+        "((FloatCeil((IntSExt((¬((2 + (EAX:32 - ECX:32))))))))₀₋₎",
+        format!("{}", expr)
+    );
```

Comment:
Do you mean for arithmetic to distinguish e.g. if multiplication is in unnecessary brackets? Or do you suggest a different style for e.g. casts?

Reply:
Well, both. Most importantly, it seems that we print each pair of brackets twice instead of once. E.g. `FloatCeil` puts brackets around its argument expression, but the argument also puts brackets around itself.